elks/arch/i86/drivers/char/console-direct-swan.c

```diff
@@ -79,24 +83,30 @@ int kraw;
 
 void soundp(unsigned period)
 {
+#ifdef CONFIG_CONSOLE_BELL
     outw(period / 12, AUD_CH1_FREQ_PORT);
     outb(0xFF, AUD_CH1_VOL_PORT);
     outb(0x01, AUD_CONTROL_PORT);
+#endif
 }
 
 void nosound(void)
 {
+#ifdef CONFIG_CONSOLE_BELL
     outb(0x01, AUD_CONTROL_PORT);
+#endif
```

Comment:
Regarding all these ifdefs - it's your port and your specific console-*swan.c file, so I am OK with all of them. IMO, getting rid of ifdefs is better, even at the cost of "slightly larger code" and using an if (setting) instead. 

Your call. Same for HALFWIDTH, etc. Someday someone may want to have all this dynamically and it gets to be rewritten.

Reply:
I do eventually want to get to the point where this is configured dynamically. However, there's an unsolved question I had before I approached that. I haven't had time to write the issue yet, but the gist of it is: "given some programs can support graphics mode and directly write to the graphics data space, which will corrupt the font and console data present therein, how - and when - should that be handled by the console driver?" It's probably a novel problem, as IBM-based platforms, as well as the PC-98, tend to separate text and graphics data...

Reply:
Ok. Since this PR is getting a bit complicated and I'm asking a lot to remove ifdefs, lets discuss the graphics/text mode issue separately. There are upcoming issues in that area for IBM PC vs PC-98 and other issues, now that we're starting to see graphics programs written for ELKS. I'm also interested in solving the problem of writing text to the "console" when graphics mode is ON.